### PR TITLE
Add validation functions

### PR DESCRIPTION
**Pull request by truizlop**
This PR adds multiple validation functions as in the equally named package in Arrow.

### CODE REVIEW

File: Sources/Bow/Data/Either.swift [hunk 1]
```diff
@@ -105,6 +105,10 @@ public class Either<A, B> : EitherOf<A, B> {
     public func combineK(_ y : Either<A, B>) -> Either<A, B> {
         return fold(constant(y), Either<A, B>.right)
     }
+    
+    public func get() -> B {
```
Comment (pakoito) [inline Sources/Bow/Data/Either.swift]: We're on the way of deprecating this. I'd suggest not adding it.

File: Sources/Bow/Validation/Validation.swift [hunk 2]
```diff
@@ -0,0 +1,475 @@
+import Foundation
+
+public class Validation<E> {
```
Comment (pakoito) [inline Sources/Bow/Validation/Validation.swift]: The validation framework is a remnant from Funktionale that's in it's way out for not being lawful. I'd also not suggest not adding it.

**Comment by truizlop** [comment]
Ok, so probably better to remove this whole branch, right?

**Comment by pakoito** [comment]
Yes, probably. I thought we had deprecation notices in place to flag it 🙈

**Pull request closed without merging**